systemtest/edu/washington/escience/myria/systemtest/JsonQuerySubmitTest.java

```diff
@@ -205,58 +210,66 @@ public void jsonQuerySubmitTest() throws Exception {
     assertEquals(((SubQueryEncoding) status.plan).fragments.size(), 3);
   }
 
-  @Test
-  public void joinChainResultTest() throws Exception {
-    // DeploymentUtils.ensureMasterStart("localhost", masterDaemonPort);
-
-    File ingestA0 = new File("./jsonQueries/multiIDB_jwang/ingest_a0.json");
-    File ingestB0 = new File("./jsonQueries/multiIDB_jwang/ingest_b0.json");
-    File ingestC0 = new File("./jsonQueries/multiIDB_jwang/ingest_c0.json");
-    File ingestR = new File("./jsonQueries/multiIDB_jwang/ingest_r.json");
-
-    HttpURLConnection conn;
-    conn = JsonAPIUtils.ingestData("localhost", masterDaemonPort, ingestA0);
-    if (null != conn.getErrorStream()) {
-      throw new IllegalStateException(getContents(conn));
+  public void jsonQuerySubmitWithResultCheck(final String[] ingests, final String[] queries, final String[] expected,
+      final String[] stored) throws Exception {
+    Preconditions.checkArgument(expected.length == stored.length);
+    for (String ingest : ingests) {
+      File ingestFile = new File(ingest);
+      HttpURLConnection conn = JsonAPIUtils.ingestData("localhost", masterDaemonPort, ingestFile);
+      if (null != conn.getErrorStream()) {
+        throw new IllegalStateException(getContents(conn));
+      }
+      assertEquals(HttpURLConnection.HTTP_CREATED, conn.getResponseCode());
+      conn.disconnect();
     }
-    assertEquals(HttpURLConnection.HTTP_CREATED, conn.getResponseCode());
-    conn.disconnect();
 
-    conn = JsonAPIUtils.ingestData("localhost", masterDaemonPort, ingestB0);
-    if (null != conn.getErrorStream()) {
-      throw new IllegalStateException(getContents(conn));
+    for (String query : queries) {
+      File queryJson = new File(query);
+      HttpURLConnection conn = JsonAPIUtils.submitQuery("localhost", masterDaemonPort, queryJson);
+      if (null != conn.getErrorStream()) {
+        throw new IllegalStateException(getContents(conn));
+      }
+      assertEquals(HttpURLConnection.HTTP_ACCEPTED, conn.getResponseCode());
+      long queryId = getQueryStatus(conn).queryId;
+      conn.disconnect();
+      while (!server.getQueryManager().queryCompleted(queryId)) {
```

Comment:
I don't really see the point of using both the JSON API and the in-process `QueryManager` API. If we're going to use the latter, then it would be simpler to wait on a `QueryFuture` rather than polling like this. If we're actually testing the JSON API, then we shouldn't cheat by using the `QueryManager` API.